docs/realm-instance.md

```diff
@@ -0,0 +1,110 @@
+# What is a Realm Instance? A Developer's Guide
```

Comment:
This probably needs an intro line to note that it's specific to Arm CCA. I know that CCA gets a mention in the first paragraph, but probably best to set expectations early on